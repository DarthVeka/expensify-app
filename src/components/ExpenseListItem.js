import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

// load a locale
numeral.register('locale', 'fr', {
    delimiters: {
        thousands: ' ',
        decimal: ','
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
    ordinal: function (number) {
        return number === 1 ? 'er' : 'ème';
    },
    currency: {
        symbol: '€'
    }
});

// switch between locales
numeral.locale('fr');

const ExpenseListItem = ({ id, description, amount, createdAt, dispatch }) => (
    <Link className='list-item' to={`/edit/${id}`}>
        <div>
            <h3>{description}</h3>
            <span>{moment(createdAt).format('MMMM Do, YYYY')}</span>
        </div>
        <h3>{numeral(amount / 100).format('$0,0.00')}</h3>
    </Link>
);

export default ExpenseListItem;