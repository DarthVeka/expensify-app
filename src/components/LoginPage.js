import React from 'react';
import { connect } from 'react-redux';
import { startGoogleLogin, startFacebookLogin, startGithubLogin } from '../actions/auth';

export const LoginPage = ({ startGoogleLogin, startFacebookLogin, startGithubLogin }) => (
    <div className='box-layout' >
        <div className='box-layout__box'>
            <h1 className='box-layout__title'>Expensify</h1>
            <p>It's time to get your expenses under control.</p>
            <button className='button button--login' onClick={startGoogleLogin} >Login with Google</button>                
            <button className='button button__facebook button--login' onClick={startFacebookLogin} >Login with Facebook</button>                
            <button className='button button__github button--login' onClick={startGithubLogin} >Login with Github</button>                
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startFacebookLogin: () => dispatch(startFacebookLogin()),
    startGithubLogin: () => dispatch(startGithubLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);