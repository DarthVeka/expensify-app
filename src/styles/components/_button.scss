.button {  
  padding: $s-size;
  font-size: $font-size-large;
  font-weight: 300;
  color: white;
  display: inline-block;  
  text-shadow: 0 1px 2px rgba(0, 0, 0, 0.25);
  background: $blue;
  border: none;
  box-shadow: inset 0 -2px #2a8bcc;
  text-decoration: none;
  line-height: 1;
}

.button--login {
  width: 190px;  
}

.button--link {
  background: none;
  box-shadow: none;  
}

.button--secondary {
  background: $gray;
  box-shadow: inset 0 -2px #7a7a7a;  
}

.button__facebook {
  background: $facebook;
  box-shadow: inset 0 -2px darken($facebook, 10%);
  margin-top: $s-size;
}

.button__github {
  background: $github;
  box-shadow: inset 0 -2px darken($github, 10%);
  margin-top: $s-size;
}