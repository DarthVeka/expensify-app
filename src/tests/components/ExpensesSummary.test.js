import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary }  from '../../components/ExpensesSummary';

test('Should correctly render expensesSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={1300} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render expensesSummary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={20451423} />);
    expect(wrapper).toMatchSnapshot();
});