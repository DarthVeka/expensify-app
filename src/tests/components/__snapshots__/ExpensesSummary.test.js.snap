// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expensesSummary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    Expenses
     totalling 
    $204,514.23
  </h1>
</div>
`;

exports[`Should correctly render expensesSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    Expense
     totalling 
    $13.00
  </h1>
</div>
`;
